File: AngularUI/UI/src/app/core/services/language.service.ts
import { SingleResponseModel } from './../models/response/singleResponseModel';
import { apiUrl } from './../../../api';
import { ResponseModel } from './../models/response/responseModel';
import { LanguageAddModel } from './../models/language/languageAddModel';
import { Service } from './service';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { LanguageModel } from '../models/language/languageModel';
import { Observable } from 'rxjs';
import { DeleteModel } from '../models/deleteModel';
import { ListResponseModel } from '../models/response/listResponseModel';

@Injectable({
  providedIn: 'root',
})
export class LanguageService
  implements Service<LanguageAddModel, LanguageModel>
{
  constructor(private httpClient: HttpClient) {}

  add(addModel: LanguageAddModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'languages/add';
    return this.httpClient.post<ResponseModel>(newPath, addModel);
  }

  delete(deleteModel: DeleteModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'languages/delete';
    return this.httpClient.post<ResponseModel>(newPath, deleteModel);
  }

  update(updateModel: LanguageModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'languages/update';
    return this.httpClient.post<ResponseModel>(newPath, updateModel);
  }

  getById(id: number): Observable<SingleResponseModel<LanguageModel>> {
    let newPath = apiUrl + 'languages/getbyid?id=' + id;
    return this.httpClient.get<SingleResponseModel<LanguageModel>>(newPath);
  }

  getAll(): Observable<ListResponseModel<LanguageModel>> {
    let newPath = apiUrl + 'languages/getall';
    return this.httpClient.get<ListResponseModel<LanguageModel>>(newPath);
  }
}

File: flutter_ui/lib/core/services/languageService.dart
import 'dart:convert';

import 'package:flutter_ui/core/interceptors/authInterceptor.dart';
import 'package:flutter_ui/core/models/deleteModel.dart';
import 'package:flutter_ui/core/models/language/languageAddModel.dart';
import 'package:flutter_ui/core/models/language/languageModel.dart';
import 'package:flutter_ui/core/models/response/listResponseModel.dart';
import 'package:flutter_ui/core/models/response/responseModel.dart';
import 'package:flutter_ui/core/models/response/singleResponseModel.dart';
import 'package:flutter_ui/core/services/service.dart';
import 'package:flutter_ui/core/services/serviceRepository.dart';
import 'package:flutter_ui/environments/api.dart';

class LanguageService extends Service
    with ServiceRepository<LanguageAddModel, LanguageModel> {
  @override
  Future<ResponseModel> add(LanguageAddModel addModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "languages/add"),
      body: addModel.toJson(),
    );

    var jsonData = json.decode(response.body);
    return ResponseModel.fromJson(jsonData);
  }

  @override
  Future<ResponseModel> delete(DeleteModel deleteModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "languages/delete"),
      body: deleteModel.toJson(),
    );

    var jsonData = json.decode(response.body);
    return ResponseModel.fromJson(jsonData);
  }

  @override
  Future<ListResponseModel<LanguageModel>> getAll() async {
    var response = await httpClient.get(
      Uri.parse(API_URL + "languages/getall"),
    );

    var jsonData = json.decode(response.body);
    return ListResponseModel<LanguageModel>.fromJson(jsonData);
  }

  @override
  Future<SingleResponseModel<LanguageModel>> getById(int id) async {
    var response = await httpClient.get(
      Uri.parse(
        API_URL + "languages/getbyid?id=" + id.toString(),
      ),
    );

    var jsonData = json.decode(response.body);
    return SingleResponseModel<LanguageModel>.fromJson(jsonData);
  }

  Future<SingleResponseModel<LanguageModel>> getByName(String name) async {
    var response = await httpClient.get(
      Uri.parse(
        API_URL + "languages/getbyname?name=" + name,
      ),
    );

    var jsonData = json.decode(response.body);
    return SingleResponseModel<LanguageModel>.fromJson(jsonData);
  }

  Future<SingleResponseModel<LanguageModel>> getByCode(String code) async {
    var response = await httpClient.get(
      Uri.parse(
        API_URL + "languages/getbycode?code=" + code,
      ),
    );

    var jsonData = json.decode(response.body);
    return SingleResponseModel<LanguageModel>.fromJson(jsonData);
  }

  @override
  Future<ResponseModel> update(LanguageModel updateModel) async {
    var response = await httpClient.post(
      Uri.parse(API_URL + "languages/update"),
      body: updateModel.toJson(),
    );

    var jsonData = json.decode(response.body);
    return ResponseModel.fromJson(jsonData);
  }
}
